### PR TITLE
Wired up Visible Assets UI to BraveWalletService

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Wired up Visible Assets UI to BraveWalletService API

1) Added ability to add or remove a token from the `ercTokenRegistry` list.
2) Added ability to add or remove a custom token.
3) Added ability to hide but not remove a custom token.

Will handle load performance optimizations in this issue 
https://github.com/brave/brave-browser/issues/18257

Resolves <https://github.com/brave/brave-browser/issues/18139>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/134246665-b727c114-079a-49b1-a85c-de6564d57f48.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx [hunk 1]
```diff
@@ -94,31 +117,56 @@ const EditVisibleAssetsModal = (props: Props) => {
...
   }
 
-  const onCheckWatchlistItem = (key: string, selected: boolean) => {
-    if (selected) {
-      const newList = [...selectedWatchlist, key]
-      setSelectedWatchlist(newList)
+  const isAssetSelected = (token: TokenInfo): boolean => {
+    const isUserToken = userVisibleTokensInfo.includes(token)
+    return isUserToken ? token.visible ?? false : false
+  }
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx:147]: Is this a safe operation in JS? Would prefer using this instead:

```js
const isUserToken = userVisibleTokensInfo.map(e => e.contractAddress).includes(token.contractAddress)
```

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx [hunk 2]
```diff
@@ -94,31 +117,56 @@ const EditVisibleAssetsModal = (props: Props) => {
...
+    setIsLoading(true)
   }
 
-  const onCheckWatchlistItem = (key: string, selected: boolean) => {
-    if (selected) {
-      const newList = [...selectedWatchlist, key]
-      setSelectedWatchlist(newList)
+  const isAssetSelected = (token: TokenInfo): boolean => {
+    const isUserToken = userVisibleTokensInfo.includes(token)
+    return isUserToken ? token.visible ?? false : false
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx:146]: How about something like this?

```js
return isUserToken && token.visible ?? false
```

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx [hunk 3]
```diff
@@ -94,31 +117,56 @@ const EditVisibleAssetsModal = (props: Props) => {
...
+  }
+
+  const isCustomToken = React.useCallback((token: TokenInfo): boolean => {
+    const assetListContracts = fullAssetList.map((token) => token.contractAddress)
+    if (token.contractAddress === 'eth') {
+      return false
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx:152]: Can we instead rely on `token.isErc20`?

File: components/brave_wallet_ui/constants/types.ts [hunk 4]
```diff
@@ -353,6 +353,7 @@ export interface TokenInfo {
   decimals: number
   visible?: boolean
   icon?: string
+  logo?: string
```
Comment (onyb) [inline components/brave_wallet_ui/constants/types.ts:356]: Why add `logo` when there's an `icon` field already?

**Review by onyb — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx]: Done!

File: components/brave_wallet_ui/constants/types.ts (hunk 4 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/constants/types.ts:358]: Logo was introduced in a separate PR and is required to be passed as an arg to the BraveWalletService api. So i needed to add it there. When we introduce token images, we will need to refactor everywhere that is using `icon` and replace it with `logo`.

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx]: Done!

**Review by onyb — APPROVED** [review]

**Pull request merged**